e2e/gke_test.go

```diff
@@ -0,0 +1,70 @@
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/gruntwork-io/terratest/modules/terraform"
+	"github.com/stretchr/testify/require"
+
+	"github.com/castai/terraform-provider-castai/castai/sdk"
+)
+
+func TestTerraformGKEOnboarding(t *testing.T) {
+	terraformOptions := terraform.WithDefaultRetryableErrors(t, &terraform.Options{
+		TerraformDir: "./tests/gke_cluster_zonal",
+	})
+
+	r := require.New(t)
+	ctx := context.Background()
+	defer func() {
+		if !t.Failed() {
```

Comment:
why destroy only on success? 🤔

Reply:
It is easier to debug if you don't need to recreate the whole cluster (it takes 30-40 minutes for whole tests to be executed)